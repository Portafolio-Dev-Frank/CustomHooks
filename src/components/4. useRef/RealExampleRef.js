import React, { useState } from 'react'
import '../2. useEffect/effects.css'
import { MultipleCustomHooks } from '../3. examples/MultipleCustomHooks'

export const RealExampleRef = () => {

    const [show, setShow] = useState(false)

    return (
        <div>
            <h1>RealExampleRef</h1>
            <hr></hr>
            {show && <MultipleCustomHooks></MultipleCustomHooks>}
            <button className="btn btn-secondary mt-5" onClick={() => setShow(!show)}>ver/ocultar</button>

        </div>
    )
}
