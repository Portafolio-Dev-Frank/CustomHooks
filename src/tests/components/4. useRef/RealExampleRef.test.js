import { shallow } from "enzyme"
import { RealExampleRef } from "../../../components/4. useRef/RealExampleRef"

describe('pruebas en <RealExampleRef></RealExampleRef> ', () => {
    test('debe renderizarse el snapshot ', () => {
        const wrapper = shallow(<RealExampleRef></RealExampleRef>)
        expect(wrapper).toMatchSnapshot();

    })

    test('debe mostrarse el componente <MultipleCustomHooks></MultipleCustomHooks> ', () => {
        const wrapper = shallow(<RealExampleRef></RealExampleRef>)
        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    })


})
