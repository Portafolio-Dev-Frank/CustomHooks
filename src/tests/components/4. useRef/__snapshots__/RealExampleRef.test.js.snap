// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <RealExampleRef></RealExampleRef>  debe renderizarse el snapshot  1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-secondary mt-5"
    onClick={[Function]}
  >
    ver/ocultar
  </button>
</div>
`;
